h/streamer/messages.py

```diff
@@ -140,8 +147,8 @@ def _generate_annotation_event(message, socket, annotation, user_nipsad):
     base_url = socket.registry.settings.get('h.app_url',
                                             'http://localhost:5000')
     links_service = LinksService(base_url, socket.registry)
-    serialized = presenters.AnnotationJSONPresenter(annotation,
-                                                    links_service).asdict()
+    resource = AnnotationResource(annotation, group_service)
```

Comment:
We aren't taking advantage of the caching mechanism of the group service if we create it here, we would effectively load the same group once for each connected client, instead of just loading and caching it once, and then reusing it.